eurocropsml/dataset/utils.py

```diff
@@ -60,7 +63,7 @@ def _split_dataset(
 
 
 def _filter_classes(
-    file_list: list[str],
+    file_list: set,
```

Comment:
Also here, the argument name contradicts the type annotation. Also can we make the annotation more specific? A set of what do we expect here? (Previously it was strings, is there a reason why this was removed?)